scope.py

```diff
@@ -0,0 +1,319 @@
+
+    :param config_wildcards:
+    :return:
+    """
+    path = pathlib.Path(__file__).parent.absolute()
+
+    for config_wildcard in config_wildcards:
+        config = config_wildcard.replace("*", "")
+        # use config defaults if configs do not exist?
+        if not (path / config).exists():
+            answer = questionary.select(
+                f"{config} does not exist, do you want to use one of the following"
+                " (not recommended without inspection)?",
+                choices=[p.name for p in path.glob(config_wildcard)],
+            ).ask()
+            subprocess.run(["cp", f"{path / answer}", f"{path / config}"])
+
+        # check contents of config.yaml WRT config.defaults.yaml
+        if config == "config.yaml":
+            with open(path / config.replace(".yaml", ".defaults.yaml")) as config_yaml:
+                config_defaults = yaml.load(config_yaml, Loader=yaml.FullLoader)
+            with open(path / config) as config_yaml:
+                config_wildcard = yaml.load(config_yaml, Loader=yaml.FullLoader)
+            deep_diff = DeepDiff(config_wildcard, config_defaults, ignore_order=True)
+            difference = {
+                k: v
+                for k, v in deep_diff.items()
+                if k in ("dictionary_item_added", "dictionary_item_removed")
+            }
+            if len(difference) > 0:
+                print("config.yaml structure differs from config.defaults.yaml")
+                pprint(difference)
+                raise KeyError("Fix config.yaml before proceeding")
+
+
+class Scope:
+
+    def __init__(self):
+        # check configuration
+        with status("Checking configuration"):
+            check_configs(config_wildcards=["config.*yaml"])
+
+            self.config = load_config(
+                pathlib.Path(__file__).parent.absolute() / "config.yaml"
+            )
+
+            # use token specified as env var (if exists)
+            kowalski_token_env = os.environ.get("KOWALSKI_TOKEN")
+            if kowalski_token_env is not None:
+                self.config["kowalski"]["token"] = kowalski_token_env
+
+        # try setting up K connection if token is available
+        if self.config["kowalski"]["token"] is not None:
+            with status("Setting up Kowalski connection"):
+                self.kowalski = Kowalski(
+                    token=self.config["kowalski"]["token"],
+                    protocol=self.config["kowalski"]["protocol"],
+                    host=self.config["kowalski"]["host"],
+                    port=self.config["kowalski"]["port"],
+                )
+        else:
+            self.kowalski = None
+            # raise ConnectionError("Could not connect to Kowalski.")
+            print("Kowalski not available")
+
+    def _get_nearest_gaia(
+            self,
+            positions: Sequence[Sequence[float]],
+            catalog: str = "Gaia_EDR3",
```

Comment:
Should this be a "current_Gaia" variable and the string itself moved to a config?

Reply:
Good call, fixed.